ta4j-core/src/main/java/org/ta4j/core/indicators/averages/WildersMAIndicator.java

```diff
@@ -0,0 +1,111 @@
+
+/**
+ * Wilder's moving average indicator.
+ *
+ * The Wilder’s Moving Average indicator (Wilder’s Smoothed Moving Average ) was
+ * developed by Welles Wilder and introduced in his 1978 book, “New Concepts in
+ * Technical Trading Systems.”
+ *
+ * The WMA inherently "remembers" past values due to its recursive formula, but
+ * the initialization WMA can significantly bias early values. This bias
+ * diminishes over time as more data contributes to the calculation. See the
+ * test cases for this indicator as it skips the initial data until the
+ * calculation stabilizes.
+ *
+ * @see <a href=
+ *      "https://www.tradingview.com/script/wXtQeoOg/#:~:text=Wilder%20did%20not%20use%20the,where%20K%20%3D1%2FN.">
+ *      https://www.tradingview.com/script/wXtQeoOg/#:~:text=Wilder%20did%20not%20use%20the,where%20K%20%3D1%2FN.</a>
+ */
+public class WildersMAIndicator extends CachedIndicator<Num> {
+
+    private final Indicator<Num> indicator;
+    private final int barCount;
+
+    /**
+     * Constructor.
+     *
+     * @param indicator an indicator
+     * @param barCount  the Wilder Moving Average time frame
+     */
+    public WildersMAIndicator(Indicator<Num> indicator, int barCount) {
+        super(indicator);
+        this.indicator = indicator;
+        this.barCount = barCount;
+    }
+
+    @Override
+    protected Num calculate(int index) {
+        NumFactory numFactory = getBarSeries().numFactory();
+        Num one = numFactory.numOf(1);
+
+        Num k = one.dividedBy(numFactory.numOf(barCount));
+        Num prevWMA = indicator.getValue(0);
+
+        // Simulate extended historical data for initialization
+        if (index < barCount) {
+            // Pretend there are extra `barCount` points before the first real point
+            // Use the first actual value as a proxy for the "missing" earlier data
+            Num simulatedValue = indicator.getValue(0);
+
+            // Initialize with a simulated Simple Moving Average (SMA)
+            Num sum = numFactory.zero();
+            for (int i = 0; i <= index; i++) {
+                sum = sum.plus(indicator.getValue(i));
+            }
+
+            // Return average of available points for initialization
+            Num preResult = sum.plus(simulatedValue.multipliedBy(numFactory.numOf(barCount - index - 1)))
+                    .dividedBy(numFactory.numOf(barCount));
+
+            prevWMA = preResult;
+            return preResult;
+        }
+
+        // For the first value, initialize with the first input value
+
+        if (index > 0) {
+            prevWMA = getValue(index - 1);
+        }
+        Num currentPrice = indicator.getValue(index);
+        return currentPrice.multipliedBy(k).plus(prevWMA.multipliedBy(numFactory.numOf(1).minus(k)));
+    }
+
+    @Override
+    public int getUnstableBars() {
+        return barCount * 2;
+    }
+
+    @Override
+    public String toString() {
+        return getClass().getSimpleName() + " barCount: " + barCount;
```

Comment:
I personally use semantic toStrings in format `INDICATOR_SHORTCUT(parameters) => current value`, that simplifies debugging.

In this case it would be `"WildersMA(%s, %d)".formatted(this.indicator, this.barCount)` at least. Formatted for example as `WildersMA(SMA(5), 5)`.

Reply:
I don't disagree with you but I was just following the pattern used for all the other indicators.  Would you like to change the ones that I created then we can do a PR to change all the others?